polaris-react/src/components/Checkbox/Checkbox.module.css

```diff
@@ -1,24 +1,36 @@
 .Checkbox {
   position: relative;
-  margin: var(--p-space-025);
+  @media (--p-breakpoints-md-up) {
+    margin: var(--p-space-025);
+  }
```

Comment:
This will remove the margin for mobile breakpoints, making the checkbox fill the 20px container. 

Do we want to remove this margin for web as well? Because with this margin, the checkbox is technically 16px in an 18px container cc @heyjoethomas